stock_picking_volume/models/stock_picking.py

```diff
@@ -16,10 +16,17 @@ class StockPicking(models.Model):
         string="Volume unit of measure label", compute="_compute_volume_uom_name"
     )
 
-    @api.depends("move_lines", "move_lines.volume")
+    @api.depends("move_lines", "move_lines.volume", "move_lines.state")
     def _compute_volume(self):
         for picking in self:
-            new_volume = sum(picking.move_lines.mapped("volume"))
+            volume_by_states = defaultdict(lambda: 0)
+            for move in picking.move_lines:
+                volume_by_states[move.state] += move.volume
+            new_volume = 0
+            states = volume_by_states.keys()
+            if states != ["cancel"]:
+                volume_by_states["cancel"] = 0
```

Comment:
Add a comment that you exclude canceled lines except for canceled picking

Reply:
I find the code complex. Why not something like:
```
moves = picking.move_lines
exclude_cancel = any(m.state != "cancel" for ml in moves)
volume = 0
for move in moves:
  if move.state == "cancel" and exclude_cancel:
    continue
  volume += move.volume
```

Reply:
yes this is much nicer